sycl/plugins/level_zero/pi_level0.cpp

```diff
@@ -2416,8 +2417,13 @@ pi_result piEventsWait(pi_uint32 NumEvents, const pi_event *EventList) {
       // is signalled, then reset the fence and command list and add them to the
       // available list for ruse in PI calls.
       EventList[I]->Queue->ZeCommandListFenceMapMutex.lock();
-      EventList[I]->Queue->resetCommandListFenceEntry(
-          EventList[I]->ZeCommandList, true);
+      ze_result_t ZeResult = ZE_CALL_NOCHECK(zeFenceQueryStatus(
+          EventList[I]
+              ->Queue->ZeCommandListFenceMap[EventList[I]->ZeCommandList]));
+      if (ZeResult == ZE_RESULT_SUCCESS) {
```

Comment:
So if fence wasn't yet signaled then this entry will remain in the cache forever, right? I think we can fix this by moving the  line 2428 under this same "if", in which case it will be reset in piEventRelease.

Reply:
fixed in the next commit, re verified that this works locally.